satellite/gracefulexit/db.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gracefulexit
+
+import (
+	"context"
+	"time"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// Progress represents the persisted graceful exit progress record
+type Progress struct {
+	NodeID            storj.NodeID
+	BytesTransferred  int64
+	PiecesTransferred int64
+	PiecesFailed      int64
+	UpdatedAt         time.Time
+}
+
+// TransferQueueItem represents the persisted graceful exit queue record
+type TransferQueueItem struct {
+	NodeID          storj.NodeID
+	Path            []byte
+	PieceNum        int32
+	DurabilityRatio float64
+	QueuedAt        time.Time
+	RequestedAt     time.Time
+	LastFailedAt    time.Time
+	LastFailedCode  int
+	FailedCount     int
+	FinishedAt      time.Time
+}
+
+// DB implements CRUD operations for graceful exit service
+//
+// architecture: Database
+type DB interface {
+	// IncrementProgress increments transfer stats for a node.
+	IncrementProgress(ctx context.Context, nodeID storj.NodeID, bytes int64, transferred int64, failed int64) error
```

Comment:
nit - I think `UpdateProgress` is a better name for this. I also think `successfulTransfers` and `failedTransfers` are preferable names to `transferred` and `failed`.

Reply:
My concern with UpdateProgress is that it sounds like you are setting the values rather than incrementing.
Changed transferred and failed